openff/toolkit/topology/topology.py

```diff
@@ -2385,3 +2386,25 @@ def hierarchy_iterator(
             if hasattr(molecule, iter_name):
                 for item in getattr(molecule, iter_name):
                     yield item
+
+    def __getattr__(self, name: str) -> List["HierarchyElement"]:
+        """If a requested attribute is not found, check the hierarchy schemes"""
+        # Avoid attempting to process dunder methods as hierarchy scheme iterator names
+        if name.startswith("__"):
+            raise AttributeError
+
+        if name in dir(self):
+            raise HierarchyIteratorNameConflictError(
+                f"Name {name} is already defined as an attribute/method of `Topology`."
+            )
+
+        try:
+            return [
+                element
+                for molecule in self.molecules
+                for element in getattr(molecule, name)
+            ]
```

Comment:
(blocking) This runs the risk of forwarding the entire Molecule API to Topology, giving us some weird stuff that we didn't intend for iteration like `Topology.partial_charges`. This should make it so only HierarchySchemes get forwarded to the Topology API. 

```suggestion
            return [
                element
                for molecule in self.molecules
                for element in molecule.hierarchy_schemes[name].hierarchy_elements)
            ]
```